### PR TITLE
Add --reference-lock-file to mirror nix 1.15.0 feature

### LINKED ISSUE #209: flakes: support the new nix feature --reference-lock-file
This would solve #144 for me, anyway, in a nice way. I can use `nix flake lock` to manipulate a temp lock file and then pass that to `nix-eval-jobs`.

This was added to nix in https://github.com/NixOS/nix/pull/8042

### PR DESCRIPTION
**Pull request by colemickens**
Fixes #209 by adding `--reference-lock-file` to the CLI.

Still testing, so draft.

### CODE REVIEW

File: src/nix-eval-jobs.cc [hunk 1]
```diff
@@ -101,6 +102,11 @@ struct MyArgs : MixEvalArgs, MixCommonArgs {
         addFlag({.longName = "flake",
                  .description = "build a flake",
                  .handler = {&flake, true}});
+        
+        addFlag({.longName = "reference-lock-file",
```
Comment (Mic92) [inline src/nix-eval-jobs.cc:106]: Looks good. Please also update the README.

File: src/nix-eval-jobs.cc [hunk 2]
```diff
@@ -43,6 +43,7 @@ using namespace nlohmann;
 struct MyArgs : MixEvalArgs, MixCommonArgs {
     std::string releaseExpr;
     Path gcRootsDir;
+    Path referenceLockFilePath;
```
Comment (Mic92) [inline src/nix-eval-jobs.cc:46]:
```suggestion
    std::optional<std::string> referenceLockFilePath;
```

Comment (Mic92) [reply src/nix-eval-jobs.cc:46]: Actually just see https://github.com/colemickens/nix-eval-jobs/pull/1/files instead.

**Comment by Mic92** [comment]
macos build failures are unrelated.

**Comment by colemickens** [comment]
I added your commit. Initial testing... is not working though.

```
[cole@zeph:~/code/nixcfg]$ nix-eval-jobs --flake .#toplevels.zeph --reference-lock-file ./flake.test.lock
warning: unknown setting 'allowed-users'
warning: unknown setting 'trusted-users'
warning: `--gc-roots-dir' not specified
warning: Git tree '/home/cole/code/nixcfg' is dirty
error: derivation must have valid outputs: error:
              … while calling the 'getAttr' builtin

                at //builtin/derivation.nix:19:19: (source not available)

              … while calling the 'derivationStrict' builtin

                at //builtin/derivation.nix:9:12: (source not available)

              (stack trace truncated; use '--show-trace' to show the full trace)

              error: getting status of '/nix/store/7p9mlmjkk51rc72v0640xlwmx4hv0qay-source/flake.lock': No such file or directory
{"attr":"","attrPath":[],"error":"error: derivation must have valid outputs: error:\n              … while calling the 'getAttr' builtin\n\n                at //builtin/derivation.nix:19:19: (source not available)\n\n              … while calling the 'derivationStrict' builtin\n\n                at //builtin/derivation.nix:9:12: (source not available)\n\n              (stack trace truncated; use '--show-trace' to show the full trace)\n\n              error: getting status of '/nix/store/7p9mlmjkk51rc72v0640xlwmx4hv0qay-source/flake.lock': No such file or directory"}

```

**Comment by Mic92** [comment]
This pull request seems to be stalled. Please re-open if you intend to continue working on this.

**Pull request closed without merging**